apps/remix-ide/src/app/files/fileManager.ts

```diff
@@ -747,50 +751,49 @@ class FileManager extends Plugin {
     }
   }
 
-  setBatchFiles (filesSet, fileProvider, override, callback) {
+  async setBatchFiles(filesSet, fileProvider, override, callback) {
     const self = this
-    if (!fileProvider) fileProvider = 'browser'
+    if (!fileProvider) fileProvider = 'workspace'
     if (override === undefined) override = false
-
-    async.each(Object.keys(filesSet), (file, callback) => {
+    console.log('setBatchFiles', filesSet, fileProvider, override)
+    for (const file of Object.keys(filesSet)) {
       if (override) {
         try {
-          self._deps.filesProviders[fileProvider].set(file, filesSet[file].content)
+          await self._deps.filesProviders[fileProvider].set(file, filesSet[file].content)
         } catch (e) {
-          return callback(e.message || e)
+          callback(e.message || e)
         }
-        self.syncEditor(fileProvider + file)
-        return callback()
-      }
+        await self.syncEditor(fileProvider + file)
+      } else {
 
-      helper.createNonClashingName(file, self._deps.filesProviders[fileProvider],
-        (error, name) => {
-          if (error) {
-            this.call('notification', 'alert', {
-              id: 'fileManagerAlert',
-              message: 'Unexpected error loading file ' + file + ': ' + error
-            })
-          } else if (helper.checkSpecialChars(name)) {
-            this.call('notification', 'alert', {
-              id: 'fileManagerAlert',
-              message: 'Special characters are not allowed in file names.'
-            })
-          } else {
-            try {
-              self._deps.filesProviders[fileProvider].set(name, filesSet[file].content)
-            } catch (e) {
-              return callback(e.message || e)
+        helper.createNonClashingName(file, self._deps.filesProviders[fileProvider],
```

Comment:
We can replace `createNonClashingName` with `createNonClashingNameAsync` for uniformity/consistency.